### DISCUSSION TITLE
[Question][Hw2] Using scapy to retrieve MAC address

### DISCUSSION THREAD

**Discussion by Standchen** [prosyslab-classroom/cs348-information-security#502] [Q&A]
Evaluation criteria says, "If your attack implements your attack packets by writing their each bytes without using libraries (e.g.,
scapy, dnspython) for all or partial attacks, you earn up to +10 extra points. This extra point is
awarded even if your total points earned is above 100 points."

Currently I'm working on the assignment by assemblying the attack packet by writing their each bytes, but I need to find some MAC address by IP. (I could patch it by hand if the script is supposed to run only on my machine, but since the script will be run for evaluation on other machine, there's a need to acquire MAC address automatically.)

Thus, my current solution for this is to utilize 'getmacbyip' function of  'scapy' library, to acquire MAC address for given IP.
I don't think using 'scapy' for this purpose isn't the case of making attack packet with "highly automated tool", but I'm afraid that such use of 'scapy' would affect my opportuinity for extra credits.

In short: Is using 'scapy' just for retrieving MAC address violation for extra credits?
